frodo.js

```diff
@@ -1,23 +1,22 @@
 /**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.